### PR TITLE
Feature APP

### PR DESCRIPTION
**Pull request by tufoinnkuo10**
In this milestone, I did the following;
- created a SPA using the create react app
- A home page with a list of filtered items when clicked takes you to the details page
- All pages have a single route and the details page has a back button to the home page
- Used the Narrative API(covid)
- Data retrieved from API is stored in the redux store
- Followed the Behance original design by Nelson Sakwa
- Add test and snapshot test for components
- Deployed with Netlify
- more on the loom video below ⬇️ 
- loom video link: https://loom.com/share/c6e59c198f114a5893fbc2eeb809ee40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,73 +26,65 @@ A simple React app that collects and displays books
...
-### `npm start`
-
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+- External API
 
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+# NETLIFY
+> [LIVE PREVIEW](https://space-ships-and-missions.netlify.app/)
```
Comment (DeliceLydia) [inline README.md]: - [x] Please add the right link

File: README.md [hunk 2]
```diff
@@ -24,73 +26,65 @@ A simple React app that collects and displays books
...
-
 This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
 
 ### Deployment
-
 This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
 
 ### `npm run build` fails to minify
-
 This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
```
Comment (DeliceLydia) [inline README.md]: - [x] I suggest that you remove these scripts

File: src/Components/Country.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { BiChevronRightCircle } from 'react-icons/bi';
+import countryName from './CountryName';
+import '../sectionStyles/acountry.css';
+
+// Country.propTypes = {country: PropTypes.string.isRequired,
```
Comment (DeliceLydia) [inline src/Components/Country.js]: - [x] remove commented code

File: src/Components/App.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  const countries = useSelector((state) => state.covidReducer.countries);
+  const total = useSelector((state) => state.covidReducer.total);
+  const load = useSelector((state) => state.covidReducer.loading);
+  const currentCntry = useSelector((state) => state.covidReducer.currentCountry);
+  const isoName = countryNames[currentCntry];
+
+  const mapImg = isoName ? `https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${isoName.toLowerCase()}/vector.svg` : '/World_map.png';
+  const dispatch = useDispatch();
+
+  // const setleclick = firstcountry
```
Comment (DeliceLydia) [inline src/Components/App.js]: - [x] Remove commented code

File: src/__Test__/apitest/CountryData.test.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { fetchCountriesFromApi } from '../../apis/CountryData';
+
+describe('fetching from api', () => {
+  // test('data type', async () => {
+  //  const data = await fetchCountriesFromApi();
+  // expect(data).toBeInstanceOf(Object);
+  //  });
```
Comment (DeliceLydia) [inline src/__Test__/apitest/CountryData.test.js]: - [x] Remove these commented codes

File: src/index.js [hunk 6]
```diff
@@ -1,17 +1,22 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import { Provider } from 'react-redux';
+// import { Browser, router } from 'react-router-dom'
```
Comment (DeliceLydia) [inline src/index.js]: - [x] Same here

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @tufoinnkuo10, 

Good job so far! my major highlights for your project:
- I liked that you came up with the idea about COVID-19 cases
- Application is working as expected

However, there are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

- [ ] Please record another video and remember to highlight a piece of code you finds interesting or you are proud of making
- [ ] The maximum length of the video is 5 minutes feel free



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
Hi @tufoinnkuo10 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada:

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:
-----------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**